lrlex/src/lib/parser.rs

```diff
@@ -750,34 +737,54 @@ mod test {
 
     #[test]
     fn test_broken_rule_names_with_spaces() {
-        let src = r#"%%
-a " "
-"#;
-        LRNonStreamingLexerDef::<DefaultLexeme<u8>, u8>::from_str(src).expect_error_at_line_col(
-            src,
-            LexErrorKind::InvalidName,
-            2,
-            3,
-        );
+        let srcs = [r#"a ""#, "a '", r#"a '""#, r#"a "'"#];
+        for line_two in srcs {
+            let src = format!("%%\n{line_two}");
+            eprintln!("lexing: {src} ");
```

Comment:
I think we need to remove this?

Reply:
I don't mind removing them if you'd like I left them for 2 reasons

1. Cargo by default will capture `stderr` not displaying it unless you run `cargo test -- --nocapture` (IIRC)
    or if the test fails in which it displays a portion of the scrollback, so you can see the src in question.
2. Since we currently aren't actually printing the source line where failure occurrs.
    and this runs through multiple source files in a loop when it fails it is currently it can be difficult to see which
    item in the array actually caused the failure.

So, I'd left it in because I thought it was harmless, and helpful.  But I could remove it and file an issue for printing the source lines on error instead, which would probably be a better fix.

Reply:
I quite often use `--nocapture` when I'm debugging, so I've developed an allergy to unnecessary chattiness :) If/when a test fails, I think it's reasonable to expect a developer to have to put various `print`s in to work out what's going on, but we can leave it to that future person :)

Reply:
Aha yes, I do see how that could be annoying then, will remove.